### ISSUE TITLE
Error generating Objective-C binding files

### ISSUE THREAD

**Issue by brianquinlan** [dart-lang/native#1269]
As of [cd8eddf8ad5b953127acf884f2b8a614482a8b0f](https://github.com/dart-lang/native/commit/cd8eddf8ad5b953127acf884f2b8a614482a8b0f), [this objective-c binding code](https://pastecode.io/s/paet5ew8) is generated and contains these compilation errors:

```
Semantic Issue (Xcode): Redefinition of 'wrapListenerBlock_ObjCBlock_ffiVoid_NSArray'
/Users/bquinlan/dart/http/pkgs/cupertino_http/src/native_cupertino_bindings.m:102:15


Semantic Issue (Xcode): Unknown type name 'NSURLHandle'
/Users/bquinlan/dart/http/pkgs/cupertino_http/src/native_cupertino_bindings.m:434:42


Semantic Issue (Xcode): Unknown type name 'NSURLHandle'
/Users/bquinlan/dart/http/pkgs/cupertino_http/src/native_cupertino_bindings.m:436:48


Semantic Issue (Xcode): Unknown type name 'NSURLHandle'
/Users/bquinlan/dart/http/pkgs/cupertino_http/src/native_cupertino_bindings.m:443:42


Semantic Issue (Xcode): Unknown type name 'NSURLHandle'
/Users/bquinlan/dart/http/pkgs/cupertino_http/src/native_cupertino_bindings.m:445:48


Semantic Issue (Xcode): Unknown type name 'NSURLHandle'
/Users/bquinlan/dart/http/pkgs/cupertino_http/src/native_cupertino_bindings.m:452:42


Semantic Issue (Xcode): Unknown type name 'NSURLHandle'
/Users/bquinlan/dart/http/pkgs/cupertino_http/src/native_cupertino_bindings.m:454:48
```

**Comment by liamappelbe** [comment]
Oops, forgot to use a `UniqueNamer` for the function names. There's two `wrapListenerBlock_ObjCBlock_ffiVoid_NSArray`s.

For the unknown type name error, I guess there's a missing import. You might be able to work around it by adding the header that declares `NSURLHandle` as an entry point. Otherwise I'll probably need to add a preamble option to the config, like we have for the Dart bindings.

**Comment by brianquinlan** [comment]
[`NSURLHandle` is deprecated](https://developer.apple.com/documentation/foundation/nsurlhandle?language=objc) - is there a way to say that I don't care about any methods that use it?

**Comment by liamappelbe** [comment]
Method filtering is on the todo list: https://github.com/dart-lang/native/issues/251. I'll try to land it in the next version

**Comment by liamappelbe** [comment]
> Oops, forgot to use a `UniqueNamer` for the function names. There's two `wrapListenerBlock_ObjCBlock_ffiVoid_NSArray`s.

Actually it's a bit more complicated than this. The function *is* being unique named (automatically, after `addDependencies`), but only its Dart name is being deduped. There's no facility to rename the C-side name (the `originalName`) in ffigen, since we've never had to do this before. There's no clean fix for this atm. I can hack something together or we can wait for the transformer refactor (#1259) for a clean fix.

**Comment by brianquinlan** [comment]
I'm going to be out for 3 weeks soon so no hurry from my POV.

**Comment by liamappelbe** [comment]
The `UniqueNamer` part of this issue was fixed in https://github.com/dart-lang/native/pull/1339. For the other part of the issue, I'll investigate automatically ignoring deprecated APIs (https://github.com/dart-lang/native/issues/1338), and also implement method filtering (https://github.com/dart-lang/native/issues/251). These two fixes will be the focus of ffigen 13.1.0
